packages/tadviewer/src/components/DataGrid.tsx

```diff
@@ -638,7 +652,13 @@ export interface DataGridProps {
     column: number,
     dataRow: DataRow,
     columnId: string,
-    cellVal: any,
+    cellVal: any
+  ) => void;
+  onGridSelectionChange?: (
+    anchor: { row: number; column: number },
+    focus: { row: number; column: number },
+    columnId: string,
+    cellVal: any
```

Comment:
I like making the callback interface explicit here, but it seems like this API will only provide value / cell data for a single cell.  That feels like too much of a special case for an API that is capable of providing events for rectangular selections.  Could you generalize this to provide an array of `columnId`, and make an array of arrays for the cell data?  Take a look at `copySelectedRange` in `DataGrid.tsx` for an example of extracting a sub-rectangle of data based on the selection. (Note: That particular function turns the data into strings for the clipboard; you will not want to do that!).
Creating a copy of the cell data whenever the selection changes is a little bit inefficient, but we can probably live with it for now.

Reply:
Thanks for this feedback - I've updated the API to take an array of columns, and array of item values.